docs/reuse.md

```diff
@@ -0,0 +1,315 @@
+          security-team.fr.red-chesterfield.com
+    
+    On-Premises
+      *.on-prem.red-chesterfield.com
+        Bare Metal Deployment
+        Air-gapped Registry
+        Static Networking
+        Developer Forks
+          lab-env.on-prem.red-chesterfield.com
+          staging.on-prem.red-chesterfield.com
+```
+
+## Project Requirements
+
+### Core Concept: Fork-Based Multi-Project Architecture
+
+**Bootstrap/ACME Base Project:**
+- Domain: `*.red-chesterfield.com` (e.g., `*.base.red-chesterfield.com`)
+- "Just works" vanilla installation with core components (Vault, Argo, Tekton)
+- Shared patterns, documentation, and common infrastructure code
+- Always functional as standalone deployment
+
+**Four Project Forks:**
+1. **Reference Implementation**: `*.ri.red-chesterfield.com` - Uses TRex as example payload
+2. **GCP HCP**: `*.gcp.red-chesterfield.com` - Google Cloud version with DNS delegation
+3. **FedRAMP Moderate**: `*.fr.red-chesterfield.com` - Compliance-focused deployment
+4. **On-Premises**: `*.on-prem.red-chesterfield.com` - Pre-configured on-prem clusters
+
+### Key Design Principles
+
+1. **"Just Works" Philosophy**: Base and any fork should deploy successfully with minimal configuration
+2. **Domain Isolation**: Each project has unique DNS wildcard domain for complete isolation
+3. **Component Replaceability**: Any infrastructure component can be removed/replaced per fork
+4. **Upstream Compatibility**: All forks should easily rebase to new Bootstrap versions
+5. **Configuration-Driven**: Small config changes enable fork-specific behavior
+6. **GitOps Native**: All changes managed through existing GitOps ApplicationSet structure
+
+## Architectural Approach
+
+### Repository Structure
+```
+bootstrap/                              # Upstream "just works" base
+├── bases/                             # Shared infrastructure components
+│   ├── clusters/                      # Cluster deployment templates
+│   └── pipelines/                     # CI/CD pipeline templates
+├── operators/                         # Core operators (Vault, GitOps, ACM, etc.)
+├── gitops-applications/global/        # Base ApplicationSets
+├── overrides/                         # NEW: Fork-specific override patterns
+│   ├── README.md                      # Override documentation and patterns
+│   ├── examples/                      # Example override implementations
+│   │   ├── remove-vault/              # Replace Vault with alternative
+│   │   ├── aws-to-gcp/                # AWS → GCP infrastructure changes
+│   │   ├── fedramp-compliance/        # FedRAMP-specific modifications
+│   │   └── domain-config/             # Domain-specific configurations
+│   └── templates/                     # Reusable override templates
+├── bootstrap.config                   # Fork-specific configuration (gitignored)
+└── bin/bootstrap                      # Enhanced with override processing
+```
+
+### Bootstrap Configuration Schema
```

Comment:
I was expecting this type of definition to be a yaml file in the repo, that the consumer modifies to meet their needs, and we provide the first one based on what we use in a given hyperscaler.